docsrc/tutorials/deploy_torch_tensorrt_to_triton.rst

```diff
@@ -0,0 +1,212 @@
+and challenges like: building an infrastructure to support concorrent model
+executions, supporting clients over HTTP or gRPC and more.
+
+The `Triton Inference Server <https://github.com/triton-inference-server/server>`__ 
+solves the aforementioned and more. Let's discuss step-by-step, the process of
+optimizing a model with Torch-TensorRT, deploying it on Triton Inference
+Server, and building a client to query the model. 
+
+Step 1: Optimize your model with Torch-TensorRT
+-----------------------------------------------
+
+Most Torch-TensorRT users will be familiar with this step. For the purpose of
+this demoonstration, we will be using a ResNet50 model from Torchhub.
+
+Let’s first pull the NGC PyTorch Docker container. You may need to create 
+an account and get the API key from `here <https://ngc.nvidia.com/setup/>`__. 
+Sign up and login with your key (follow the instructions
+`here <https://ngc.nvidia.com/setup/api-key>`__ after signing up).
+
+::
+
+   # <xx.xx> is the yy:mm for the publishing tag for NVIDIA's Pytorch 
+   # container; eg. 22.04
+
+   docker run -it --gpus all -v /path/to/folder:/resnet50_eg nvcr.io/nvidia/pytorch:<xx.xx>-py3
+
+Once inside the container, we can proceed to download a ResNet model from
+Torchhub and optimize it with Torch-TensorRT. 
+
+::
+
+   import torch
+   import torch_tensorrt
+   torch.hub._validate_not_a_forked_repo=lambda a,b,c: True
+
+   # load model
+   model = torch.hub.load('pytorch/vision:v0.10.0', 'resnet50', pretrained=True).eval().to("cuda")
+
+   # Compile with Torch TensorRT; 
+   trt_model = torch_tensorrt.compile(model, 
+       inputs= [torch_tensorrt.Input((1, 3, 224, 224))],
+       enabled_precisions= { torch.half} # Run with FP32
+   )
+
+   # Save the model
+   torch.jit.save(trt_model, "model.pt")
+
+The next step in the process is to set up a Triton Inference Server.
+
+Step 2: Set Up Triton Inference Server
+--------------------------------------
+
+If you are new to the Triton Inference Server and want to learn more, we
+highly recommend to checking our `Github
+Repository <https://github.com/triton-inference-server>`__.
+
+To use Triton, we need to make a model repository. A model repository, as the
+name suggested, is a repository of the models the Inference server hosts. While
+Triton can serve models from multiple repositories, in this example, we will
+discuss the simplest possible form of the model repository.
+
+The structure of this repository should look something like this:
+
+::
+
+   model_repository
+   |
+   +-- resnet50
+       |
+       +-- config.pbtxt
+       +-- 1
+           |
+           +-- model.pt
+
+There are two files that Triton requires to serve the model: the model itself
+and a model configuration file which is typically provided in ``config.pbtxt``.
+For the model we prepared in step 1, the following configuration can be used: 
+
+::
+
+   name: "resnet50"
+   platform: "pytorch_libtorch"
+   max_batch_size : 0
```

Comment:
Is this required ? max_batch_size=0 might be confusing

Reply:
The model we export is explicitly 1,3,224,224 (no batching) Per triton docs, max_batch_size needs to be set to zero. I have added links to the model config page in case someone has more questions

Reply:
In the (1, 3, 224, 224), `1` is the batch size. Do you know why we have 
```
input [
  {
    name: "input__0"
    data_type: TYPE_FP32
    dims: [ 3, 224, 224 ]
    reshape { shape: [ 1, 3, 224, 224 ] }
  }
]
```
dims as [3, 224,224] and not directly [1, 3, 224, 224] here ? Why is a reshape required ?

Reply:
The client is just sending a 3,224,224 image. Models input size (as exported from onnx is 1,3,224,224), hence the reshape required